design/accepted/gc_plus/gc-plus-cloud.md

```diff
@@ -0,0 +1,64 @@
+# Managed Uncommitted Garbage Collection for lakeFS cloud - Execution Plan
+
+## Goals
+1. Run UGC for lakeFS cloud users
+2. Allow Cloud users to clean up storage space by deleting objects from the underlying storage for cost efficiency
+3. Support new and existing cloud users (both repository structures)
+
+## Non-Goals
+1. Remove lakeFS limitations during the UGC job run
+2. Improve UGC OSS support
+
+### User Story
+For every cloud installation, The UGC will run as default as a background job (sweep=true) without visibility to the user to control its configuration.
+We should have to ability to control the MUGC for the user - change configuration, or stop it.
+
+### Permissions
+Create default UGC user with permissions for every new cloud env (The permissions to Enigma was added to the GC user)
+- lakeFS permissions - prepare uncommitted gc
+- Bucket permissions - role to list, write, delete (can get the same role as the GC)
+
+### Managed UGC job
+- Control plane interface (design with cloud native team)
+  - Add a step to initialize the MUGC job when creating a new cloud installation
+  - Configuration for the spark job (configured role, secret and access key, installation endpoint, repository)
+  - Backup & Restore
+- Deployment of new UGC version
+- Migration (Enigma)
+- UGC client and server compatibility
+- Run the UGC periodically (AWS lambda cron job, Airflow DAG, EMR serverless, etc.)
+- Declare SLA (might require performance improvements)
+
+### Metrics and Logging
+- Add UGC logging and metrics to the cloud monitoring tools
+- Alerti configuration / on-call guide - for failed job runs add P3/P4 alerts 
+- Access to the cloud user metadata for debugging
```

Comment:
The UGC writes to the `_lakefs` directory under the retention prefix the run report and parquet file with the uncommitted objects. My team may need access to these files and the other metadata for debugging in case of an error.